src/test/java/seedu/address/logic/commands/EditCommandTest.java

```diff
@@ -167,11 +167,23 @@ public void execute_noChangeInEditFields_throwsCommandException() {
         assertThrows(CommandException.class, () -> editCommand.execute(model), MESSAGE_NO_CHANGE);
     }
 
+    @Test
+    public void execute_emptyMedicalHistory_throwsCommandException() {
+        Person person = model.getFilteredPersonList().get(0);
+
+        EditPersonDescriptor descriptor = new EditPersonDescriptorBuilder().withMedicalHistories().build();
+        EditCommand editCommand = new EditCommand(person.getName(), null, descriptor);
+
+        assertThrows(CommandException.class, () -> editCommand.execute(model), MESSAGE_EMPTY_MEDICAL_HISTORY_TO_EDIT);
+
+    }
+
     @Test
     public void undo_successfulEditCommand() throws CommandException {
         Model model = new ModelManager();
-        Person originalPerson = new PersonBuilder().build();
-        Person editedPerson = new PersonBuilder().withName(VALID_NAME_AMY).withPhone(VALID_PHONE_BOB).build();
+        Person originalPerson = new PersonBuilder().withMedical(VALID_MEDICALHISTORY).build();
+        Person editedPerson = new PersonBuilder().withName(VALID_NAME_AMY).withPhone(VALID_PHONE_BOB)
+                .withMedical(VALID_MEDICALHISTORY).build();
```

Comment:
May I check why this change was made? Asking this because in `EditCommand.java`, I do not see a code change related.

Reply:
Thank you for this review. It is a consequence of a bug I did not realize existed after reading this comment. I will be updating the PR after fixing the bug.